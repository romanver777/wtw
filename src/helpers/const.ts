export const genres: Array<string> = [
  "Все жанры",
  "Комедии",
  "Документальное",
  "Драма",
  "Ужасы",
  "Семейное",
  "Романтические",
  "Фантастика",
  "Триллеры",
];

export const tabsName: Array<string> = ["Описание", "Информация", "Отзывы"];

export const similarMovies = "Вам может понравиться";

export const page: { [key: string]: string } = {
  movieCard: "movie-card",
  userPage: "user-page",
};

export const userPageTitle = {
  signIn: "Выполните вход",
  myList: "Мой список",
};
