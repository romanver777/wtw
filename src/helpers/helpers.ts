export const convertRatingLevel = (rt: string | null): string => {
  const rating = Number(rt);
  if (rating > 0 && rating < 4) return "Плохо";
  if (rating >= 4 && rating < 5) return "Удовлетворительно";
  if (rating >= 5 && rating < 6) return "Хорошо";
  if (rating >= 6 && rating < 7) return "Очень хорошо";
  if (rating >= 7 && rating < 8) return "Отлично";
  if (rating >= 8) return "Превосходно";
  return "-";
};

export const convertDateTime = (date: string): string => {
  return new Date(date).toLocaleString("ru-RU", {
    year: "numeric",
    month: "long",
    day: "numeric",
    hour: "numeric",
    minute: "numeric",
  });
};
