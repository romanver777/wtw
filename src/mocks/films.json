{
  "films": [
    {
      "filmId": 1302121,
      "nameRu": "Соединённые Штаты против Билли Холидей",
      "nameEn": "The United States vs. Billie Holiday",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "биография"
        },
        {
          "genre": "музыка"
        }
      ],
      "rating": "5.8",
      "ratingVoteCount": 586,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1302121.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1171895,
      "nameRu": "Неуязвимый (сериал)",
      "nameEn": "Invincible",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "ужасы"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 1549,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1171895.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1171895.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245486,
      "nameRu": "Невероятные (сериал)",
      "nameEn": "The Nevers",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "Великобритания"
        },
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 882,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245486.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1015471,
      "nameRu": "Райя и последний дракон",
      "nameEn": "Raya and the Last Dragon",
      "year": "2021",
      "filmLength": "1:47",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.6",
      "ratingVoteCount": 43386,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1015471.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1015471.jpg",
      "ratingChange": null
    },
    {
      "filmId": 571335,
      "nameRu": "Бесстыжие (сериал)",
      "nameEn": "Shameless",
      "year": "2011-2021",
      "filmLength": "0:55",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "8.6",
      "ratingVoteCount": 207954,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/571335.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/571335.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1328036,
      "nameRu": "Happy End (сериал)",
      "nameEn": null,
      "year": "2021-...",
      "filmLength": "0:35",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "7.4",
      "ratingVoteCount": 3282,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1328036.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1328036.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1045583,
      "nameRu": "Конёк-Горбунок",
      "nameEn": null,
      "year": "2021",
      "filmLength": "1:51",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "6.8",
      "ratingVoteCount": 100901,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1045583.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1045583.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1274271,
      "nameRu": "Родные",
      "nameEn": null,
      "year": "2020",
      "filmLength": "1:40",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.4",
      "ratingVoteCount": 20539,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1274271.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1274271.jpg",
      "ratingChange": null
    },
    {
      "filmId": 767379,
      "nameRu": "Фарго (сериал)",
      "nameEn": "Fargo",
      "year": "2014-...",
      "filmLength": "0:53",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.4",
      "ratingVoteCount": 119492,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/767379.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/767379.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245524,
      "nameRu": "Зависнуть в Палм-Спрингс",
      "nameEn": "Palm Springs",
      "year": "2020",
      "filmLength": "1:30",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Гонконг"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "мелодрама"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "7.2",
      "ratingVoteCount": 130542,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245524.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245524.jpg",
      "ratingChange": null
    },
    {
      "filmId": 4430018,
      "nameRu": "Абсолютное доказательство",
      "nameEn": "Absolute Proof",
      "year": "2021",
      "filmLength": "2:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "документальный"
        }
      ],
      "rating": null,
      "ratingVoteCount": null,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/4430018.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/4430018.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1437860,
      "nameRu": "Чикатило (сериал)",
      "nameEn": null,
      "year": "2020-2020",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "детектив"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "5.9",
      "ratingVoteCount": 7820,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1437860.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1437860.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1005008,
      "nameRu": "Майор Гром",
      "nameEn": null,
      "year": "2017",
      "filmLength": "0:25",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "короткометражка"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 29251,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1005008.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1005008.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1291310,
      "nameRu": "В её глазах (мини-сериал)",
      "nameEn": "Behind Her Eyes",
      "year": "2021-2021",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "детектив"
        }
      ],
      "rating": "7.3",
      "ratingVoteCount": 12949,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1291310.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1291310.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1038789,
      "nameRu": "Охотник на монстров",
      "nameEn": "Monster Hunter",
      "year": "2020",
      "filmLength": "1:43",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Китай"
        },
        {
          "country": "Германия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "5.6",
      "ratingVoteCount": 24933,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1038789.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1038789.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1248786,
      "nameRu": "Аферистка",
      "nameEn": "I Care a Lot",
      "year": "2020",
      "filmLength": "1:58",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "6.6",
      "ratingVoteCount": 81569,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1248786.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1248786.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1193013,
      "nameRu": "Грань времени",
      "nameEn": "Synchronic",
      "year": "2019",
      "filmLength": "1:42",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.2",
      "ratingVoteCount": 10361,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1193013.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1193013.jpg",
      "ratingChange": null
    },
    {
      "filmId": 958500,
      "nameRu": "Охотник за разумом (сериал)",
      "nameEn": "Mindhunter",
      "year": "2017-2019",
      "filmLength": "1:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 50186,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/958500.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/958500.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1347194,
      "nameRu": "Звездный рубеж",
      "nameEn": "Cosmic Sin",
      "year": "2021",
      "filmLength": "1:28",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "3.4",
      "ratingVoteCount": 524,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1347194.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1347194.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1236082,
      "nameRu": "Ассасин: Битва миров",
      "nameEn": "Ci sha xiao shuo jia",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "Китай"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "6.0",
      "ratingVoteCount": 645,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1236082.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1236082.jpg",
      "ratingChange": null
    }
  ]
}
