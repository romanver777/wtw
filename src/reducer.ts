import movies from "./mocks/films.json";
import { FilmsType } from "./types/types";
import { GENRES } from "./helpers/const";

interface InitStateType {
  films: FilmsType;
  genre: string;
}

interface ActionType {
  SET_GENRE: string;
  GET_FILMS_BY_GENRE: string;
}

const InitialState: InitStateType = {
  films: [...movies],
  genre: GENRES[0],
};

const ActionType = {
  SET_GENRE: `SET_GENRE`,
  GET_FILMS_BY_GENRE: `GET_FILMS_BY_GENRE`,
};

const ActionCreator = {
  setGenre: (genre: string) => ({
    type: ActionType.SET_GENRE,
    payload: genre,
  }),
  getFIlmsByGenre: (genre, films) => ({
    type: ActionCreator.GET_FILMS_BY_GENRE,
    payload: [],
  }),
};

const reducer = (state = InitialState, action): InitStateType => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };
    default:
      return state;
  }
};

const Operation = {};

export { reducer, ActionCreator, Operation };
