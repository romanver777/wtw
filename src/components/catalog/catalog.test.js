/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { reducer } from "../../reducer";
import { Router } from "react-router";
import history from "../../history";

import Catalog from "./catalog";
import films from "../../mocks/films.json";
import { ALL_GENRES } from "../../helpers/const";

const initState = {
  films,
  awaitFilm: films[2],
  currentFilm: null,
  hoveredFilm: null,
  activeGenre: ALL_GENRES,
  genresList: null,
  tk: null,
  reviews: null,
  staff: null,
  hoveredVideoData: null,
};

const store = createStore(reducer, initState);

describe("Catalog component", () => {
  const props = { pageName: "main" };
  const Trenderer = TestRenderer.create(
    <Provider store={store}>
      <Router history={history}>
        <Catalog {...props} />
      </Router>
    </Provider>
  );

  it("renders correctly", () => {
    expect(Trenderer.toJSON()).toMatchSnapshot();
  });
});
