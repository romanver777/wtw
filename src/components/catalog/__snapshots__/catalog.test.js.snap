// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog component renders correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Соединённые Штаты против Билли Холидей"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1302121"
          onClick={[Function]}
        >
          Соединённые Штаты против Билли Холидей
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Неуязвимый (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1171895.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1171895"
          onClick={[Function]}
        >
          Неуязвимый (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Невероятные (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1245486"
          onClick={[Function]}
        >
          Невероятные (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Райя и последний дракон"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1015471.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1015471"
          onClick={[Function]}
        >
          Райя и последний дракон
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Бесстыжие (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/571335.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/571335"
          onClick={[Function]}
        >
          Бесстыжие (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Happy End (сериал)"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1328036.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1328036"
          onClick={[Function]}
        >
          Happy End (сериал)
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Конёк-Горбунок"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1045583.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1045583"
          onClick={[Function]}
        >
          Конёк-Горбунок
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Родные"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1274271.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#/film/1274271"
          onClick={[Function]}
        >
          Родные
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Больше
    </button>
  </div>
</section>
`;
